lib/client/api_matrix.es6.js

```diff
@@ -387,3 +456,160 @@ foam.CLASS({
     },
   ],
 });
+
+foam.CLASS({
+  name: 'ApiMatrixController',
+  package: 'org.chromium.apis.web',
+  implements: ['foam.mlang.Expressions'],
+
+  requires: [
+    'org.chromium.apis.web.ApiMatrix',
+    'org.chromium.apis.web.Release',
+  ],
+  imports: ['releaseDAO', 'releaseWebInterfaceJunctionDAO'],
+
+  topics: ['newMatrix'],
+
+  properties: [
+    {
+      class: 'FObjectArray',
+      of: 'org.chromium.apis.web.Release',
+      name: 'releases',
+    },
+    {
+      class: 'String',
+      name: 'searchKey',
+    },
+    {
+      name: 'releaseOptions',
+      value: null,
+    },
+    {
+      name: 'numAvailable',
+      value: null,
+    },
+    {
+      class: 'FObjectArray',
+      of: 'org.chromium.apis.web.ReleaseWebInterfaceJunction',
+      name: 'data_',
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'org.chromium.apis.web.ApiMatrix',
+      name: 'apiMatrix_',
+      factory: function() { return this.ApiMatrix.create(); },
+    },
+    {
+      name: 'matrix_',
+      factory: function() { return {}; },
+    },
+    {
+      name: 'matrixPromise_',
+      value: Promise.resolve({}),
```

Comment:
Is this a good default? Wouldn't it be better to reject until a fetch has started?

Reply:
My design thinking here was that any view requesting "the current matrix" always gets a valid matrix. Until data are loaded, the current matrix is `{}`.

Reply:
So why is it a promise at all? Default to empty, swap in a new one when available.

Reply:
This allows us to write `this.matrixPromise_.then()` without having to write a guard like `if (this.matrixPromise_) ...`.